src/odemis/gui/test/comp_overlay_test.py

```diff
@@ -726,6 +727,84 @@ def on_key(evt):
 
         cnvs.Bind(wx.EVT_KEY_UP, on_key)
 
+    def test_ruler_overlay(self):
+        logging.getLogger().setLevel(logging.DEBUG)
+        cnvs = miccanvas.DblMicroscopeCanvas(self.panel)
+
+        tab_mod = self.create_simple_tab_model()
+        view = tab_mod.focussedView.value
+
+        self.add_control(cnvs, wx.EXPAND, proportion=1, clear=True)
+        cnvs.setView(view, tab_mod)
+        cnvs.current_mode = TOOL_RULER
+
+        rol = wol.RulerOverlay(cnvs)
+        rol.activate()
+        rol.enabled = True
```

Comment:
You could replace all these lines by the same thing you did for the export test cases: `tab_mod.tool.choices |= {TOOL_RULER}`. Then `rol = cnvs.ruler_overlay`.
Actually, I don't think that changing the "cnvs.current_mode" has any effect, does it?